core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -76,25 +86,65 @@ protected boolean requiresRelease() {
         return true;
     }
 
-    private ProbeResult fromSCMLink(ProbeContext context, String scm) {
+    private ProbeResult fromSCMLink(ProbeContext context, String scm, String pluginName) {
         Matcher matcher = GH_PATTERN.matcher(scm);
         if (!matcher.find()) {
             if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("{} is not respecting the SCM URL Template", scm);
+                LOGGER.debug("{} is not respecting the SCM URL Template.", scm);
             }
-            return ProbeResult.failure(key(), "SCM link doesn't match GitHub plugin repositories");
+            return ProbeResult.failure(key(), "SCM link doesn't match GitHub plugin repositories.");
         }
-
         try {
             context.getGitHub().getRepository(matcher.group("repo"));
-            return ProbeResult.success(key(), "The plugin SCM link is valid");
+            Optional<Path> pluginPathInRepository  = findPluginPom(context.getScmRepository(), pluginName, context.getRepositoryName(scm).get().split("/")[1]);
+            String folderPath = pluginPathInRepository
+                .flatMap(path -> Optional.ofNullable(path.getParent()))
+                .map(parent -> parent.getFileName().toString())
+                .orElseGet(() -> pluginPathInRepository.map(Path::toString).orElse(""));
+            context.setScmFolderPath(folderPath);
+            return ProbeResult.success(key(), "The plugin SCM link is valid.");
         } catch (IOException ex) {
-            return ProbeResult.failure(key(), "The plugin SCM link is invalid");
+            return ProbeResult.failure(key(), "The plugin SCM link is invalid.");
         }
     }
 
     @Override
     public String[] getProbeResultRequirement() {
         return new String[]{UpdateCenterPluginPublicationProbe.KEY};
     }
+
+    /**
+     * Searches for Pom file in every directory available in the repository
+     *
+     * @param directory path in the scm
+     * @param pluginName the name of the plugin
+     * @return folderPath if it is valid
+     */
+    private Optional<Path> findPluginPom(Path directory, String pluginName, String rootDirectory) {
```

Comment:
Why do you need the third parameter? Just to return it if something wrong is happening? Please don't, handle this in the calling method.